src/forge/actors/generator.py

```diff
@@ -617,3 +717,26 @@ async def validate_model_params(self, validate_fn):
         return validate_fn(
             self._debug_saved_params, self.worker.model_runner.model, logger
         )
+
+
+class _WeightFetcher(ForgeActor):
```

Comment:
I think this could be a method on the generator that gets called from main, so prefetch is controlled and visible from the main loop. I am curious if this has to actually be a separate process since this is an async method and I would think most of the time it's waiting on ts.get.